File: WebRTC/WebRTCServer.js
// signaling-server.js
const WebSocket = require('ws');
const wss = new WebSocket.Server({ port: 8080 });

wss.on('connection', function connection(ws) {
    ws.on('message', function incoming(message) {
        // Relay message to all clients except the sender
        wss.clients.forEach(function each(client) {
            if (client !== ws && client.readyState === WebSocket.OPEN) {
                client.send(message);
            }
        });
    });
});

console.log('Signaling server running on ws://localhost:8080');

File: WebRTC/Server/webrtcserver.py
import asyncio
import websockets

# Store all connected peers
peers = set()

async def signaling_server(websocket, path):
    peers.add(websocket)
    try:
        async for message in websocket:
            # Broadcast the message to all other peers
            for peer in peers:
                if peer != websocket:
                    await peer.send(message)
    finally:
        peers.remove(websocket)

async def main():
    # Start the server
    async with websockets.serve(signaling_server, "localhost", 8765):
        print("WebRTC signaling server running on ws://localhost:8765")
        await asyncio.Future()  # Run forever

# Run the main coroutine
if __name__ == "__main__":
    asyncio.run(main())
